optaplanner-test/src/main/java/org/optaplanner/test/impl/score/stream/DefaultSingleConstraintVerification.java

```diff
@@ -35,8 +46,53 @@ protected DefaultSingleConstraintVerification(
         this.scoreDirectorFactory = scoreDirectorFactory;
     }
 
+    private void assertDistinctPlanningIds(SolutionDescriptor<Solution_> solutionDescriptor, Object... facts) {
```

Comment:
As you can see, the logic is very heavy. This will slow the Constraint Verifier down.
We have to decide if this is even something we want.